hack/integration-test/setup.sh

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+: "${BUILD_DEFS_REPO:?Missing repository path of build-definitions}"
+: "${QUAY_NAMESPACE:?Missing QUAY_NAMESPACE. build-and-push.sh uses it to push task bundles}"
+
+export QUAY_NAMESPACE
+
+declare -r DEFAULT_RECIPES="
+clone 0.2 -
+clone 0.2 M
+clone 0.2 -
+clone 0.2 M
+clone 0.2 -
+lint 0.2 -
+lint 0.2 M
+lint 0.2 M
+lint 0.2 M
+lint 0.2 -
+"
+
+: "${RECIPES:=$DEFAULT_RECIPES}"
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$SCRIPTDIR" || exit 1
+
+declare -r DEFINITIONS_DIR="${SCRIPTDIR}/definitions"
+
+
+build_and_push() {
+    HERE="$DEFINITIONS_DIR" \
+    SKIP_INSTALL=1 \
+    SKIP_BUILD=1 \
+    BUILD_TAG=$(git rev-parse HEAD) \
+    OUTPUT_TASK_BUNDLE_LIST=/tmp/pmt-test-task-bundle-list \
+    OUTPUT_PIPELINE_BUNDLE_LIST=/tmp/pmt-test-pipeline-bundle-list \
+        bash "${BUILD_DEFS_REPO}/hack/build-and-push.sh"
+}
+
+bump_task_version() {
+    local -r task_file=$1
+    local patch
+    local new_version
+
+    IFS=. read -r major minor patch < <(
+        yq '.metadata.labels."app.kubernetes.io/version"' "$task_file"
+    )
+    patch=$((patch+1))
+    new_version="${major}.${minor}.${patch}"
+    # app.kubernetes.io/version: "0.3.1"
+    sed -i "s|^\( \+app\.kubernetes\.io/version\): .\+$|\1: \"${new_version}\"|" "$task_file"
+    echo "$new_version"
+}
+
+create_migration() {
+    local -r task_file=$1
+    local -r new_version=$2
+    local -r versioned_dir=${task_file%/*}
+
+    local -r migration_dir="${versioned_dir}/migrations"
+    [ -e "$migration_dir" ] || mkdir -p "$migration_dir"
+
+    local -r migration_file="${migration_dir}/${new_version}.sh"
+
+    cat >"$migration_file" <<EOF
+#!/usr/bin/env bash
+set -euo pipefail
+declare -r pipeline_file=\${1:?Missing pipeline file}
+
+# Created at $(date)
+# Migration for task ${task_file##*/} and version ${new_version}
```

Comment:
The migration scripts don't make any edits to the pipeline, which can be confusing. I ran `pipeline-migration-tools` test expecting to see some diff, but there is none.